extensions/amp-image-viewer/0.1/amp-image-viewer.js

```diff
@@ -261,104 +245,121 @@ export class AmpImageViewer extends AMP.BaseElement {
   }
 
   /**
-   * @return {number}
+   * Sets the source width and height based on the user-defined
+   * dimensions of the amp-img if exists, the natural dimensions
+   * of the source image if loaded, and the offset dimensions of
+   * amp-img element if not.
+   * @param {!Element}
    * @private
    */
-  getSourceWidth_() {
-    if (this.sourceAmpImage_.hasAttribute('width')) {
-      return parseInt(this.sourceAmpImage_.getAttribute('width'), 10);
+  setSourceDimensions_(ampImg) {
+    let width, height;
+    if ((width = ampImg.getAttribute('width'))
```

Comment:
Reassigning the mutable `width` and `height` variables in `if` conditions makes this section kind of confusing. I think it could be a little cleaner with a few separate pairs of `const`s